src/sentry/notifications/helpers.py

```diff
@@ -732,134 +729,102 @@ def get_query(
-        user_ids: The user ids to get notification settings for.
+        recipients: The recipients of the notification settings (user or team).
+        projects_ids: The projects to get notification settings for.
+        organization_id: The organization to get notification settings for.
+        additional_filters: Additional filters to apply to the query for NotificationSettingProvider.
     """
-    query = get_query(recipients, project, organization)
-    return NotificationSettingOption.objects.filter(query)
-
-
-def get_setting_options_for_recipient(
-    recipient: RpcActor | Team | RpcUser,
-    project: Project | None = None,
-    organization: Organization | None = None,
-) -> MutableMapping[NotificationSettingEnum, NotificationSettingsOptionEnum]:
-    all_settings = get_all_setting_options([recipient], project, organization)
-
-    notification_settings = {}
-    # Project settings take precedence over all other notification settings
-    for setting in all_settings:
-        if setting.scope_type == NotificationScopeEnum.PROJECT.value:
-            notification_settings[
-                NotificationSettingEnum(setting.type)
-            ] = NotificationSettingsOptionEnum(setting.value)
-
-    # Organization settings apply when project settings are not set
-    for setting in all_settings:
-        if setting.scope_type == NotificationScopeEnum.ORGANIZATION.value:
-            setting_type = NotificationSettingEnum(setting.type)
-            if setting_type not in notification_settings:
-                notification_settings[setting_type] = NotificationSettingsOptionEnum(setting.value)
-
-    # Team/User settings are the most specific.
-    if recipient_is_user(recipient):
-        scope_type = NotificationScopeEnum.USER.value
-    elif recipient_is_team(recipient):
-        scope_type = NotificationScopeEnum.TEAM.value
-    else:
-        raise Exception("recipient must be either user or team")
+    if not additional_filters:
+        additional_filters = Q()
 
-    for setting in all_settings:
-        if setting.scope_type == scope_type:
-            setting_type = NotificationSettingEnum(setting.type)
-            if setting_type not in notification_settings:
-                notification_settings[setting_type] = NotificationSettingsOptionEnum(setting.value)
+    query = get_query(recipients, project_ids, organization_id)
+    return NotificationSettingProvider.objects.filter(query & additional_filters)
 
-    # Fill in any missing settings with the default
-    defaults = get_type_defaults()
-    for type in NotificationSettingEnum:
-        if type not in notification_settings and type in defaults:
-            notification_settings[type] = defaults[type]
 
-    return notification_settings
+def get_recipient_from_team_or_user(user_id: int | None, team_id: int | None) -> RpcUser | Team:
+    if user_id is not None:
+        recipient = RpcUser(id=user_id)
```

Comment:
Does this recipient get any of its other attributes used? You can use `user_service.get_user()` to fetch the hydrated user record.